.github/workflows/inspektor-gadget.yml

```diff
@@ -6,8 +6,7 @@ env:
 concurrency:
   group: ${{ github.ref }}
   cancel-in-progress: true
-on:
-  push:
+on: [push, pull_request]
```

Comment:
Sometimes I like the fact that we can run the CI without opening a PR (to avoid spamming while fixing something with the CI), however I think it's perfectly fine to limit this and only run when there are pushes to the main branch **and releases**.

In summary, I think something like should work fine: 

```yaml
on:
  push:
    branches:
    - master
    - v* <-- we can use a more specific filter here 
  pull_request:
    branches:
    - master <-- no sure about this one. What if we to open a PR against a dev branch? 
```

Reply:
I think (not sure, we have to try) that `pull_request` without any filtering will cover PRs against any branch from internal and external contributions. Then, `push` on `master` and tag `v` will cover the merges to main and releases:

```yaml
on:
  pull_request:
  push:
    branches:
    - master
    - v* <-- we can use a more specific filter here 
```

Reply:
Notice that with such filtering on `push`, the workflow will not be triggered when we push a commit without creating a PR. I don't know, in some cases, it is useful because maybe we just want to share it. In others, it is not. For instance, I usually push my changes and wait until the CI is green before opening the PR to keep the PR history clean. But I don't think it is huge problem. WDYT?

Reply:
I tested all these cases in a fork, and it worked as I described. The workflow was triggered in the following situations:
- PR against main and dev branch
- Merge into `main`
- Tags starting with `'v'`

 I still need to test the external contributions. Could any of you open a PR against [my fork](https://github.com/blanquicet/inspektor-gadget).

Reply:
> With this modification, it means each time we will push on an already opened PR (e.g. to address some review comments) the CI will not be run again?

I think that's not the case. CI won't run for branches without having a PR. (If you push to mauricio/test-me it won't be run by the CI until you open the PR for it). 

> the workflow will not be triggered when we push a commit without creating a PR.

I also like to push and wait for CI to run but I think it's fine to disable it, also because many times the CI runs for branches we don't intent to open PRs for.

Reply:
I confirm what Mauricio said. Sorry if my explanation wasn't clear enough. 

I validated the external contribution:

![immagine](https://user-images.githubusercontent.com/57147708/186184435-706fa397-2d89-41c9-a225-50e33eb1d948.png)

Reply:
> With this modification, it means each time we will push on an already opened PR (e.g. to address some review comments) the CI will not be run again?
>

@eiffel-fl No. If you force push on https://github.com/blanquicet/inspektor-gadget/pull/5, it will indeed trigger the CI.